Cargo.toml

```diff
@@ -19,13 +19,18 @@ maintenance = { status = "actively-developed" }
 
 [features]
 default = ["sink"]
+
+# Whether publishing is enabled
+publish = []
+
 # Publishers
 google = ["base64", "yup-oauth2", "hyper", "http", "serde_json", "serde", "serde/derive", "uuid/serde"]
```

Comment:
Won't the specific publishing implementations also depend on the overall publish feature?

Reply:
that's a fair question. I was envisioning that the "google" feature covers both the pubsub publisher and consumer, so that users only toggle one feature. The actual Publisher and Consumer implementations would still be under the `publish` and `consume` modules, so they can depend on publish/consume functionality, but any uses would be coupled to enabling those features anyway and wouldn't fail to compile if they're absent.

This single feature does not, however, enable cutting dependencies if e.g. the publisher and consumer use different deps -- and concretely we will have this for google as long as the publisher is REST and the consumer is gRPC. We could use separate features like `google-publish` + `google-consume`, but I think that this dependency division is an exception. Practically any future implementation like AWS would just have the same library dependency for publishing and consuming, as will google once we shift to gRPC for the publisher too